### DISCUSSION TITLE
Named default import of `next/document` breaks build

### DISCUSSION THREAD

**Discussion by harrysolovay** [vercel/next.js#14543] [Help]
Inside of `_document.tsx`, the following produces a build error:

```diff
import * as React from "react";
import {
  Head,
  Html,
  Main,
+  default as NextDocument,
  NextScript,
} from "next/document";
```

Feels as though it should work the same as this (successfully builds):

```diff
import * as React from "react";
+ import Document, {Head, Html, Main, NextScript} from "next/document";
```

Ran into this:

`RangeError: Maximum call stack size exceeded

Error occurred prerendering page "/404". Read more: https://err.sh/next.js/prerender-error
RangeError: Maximum call stack size exceeded

> Build error occurred
Error: Export encountered errors on following paths:
        /404
    at exportApp (/Users/harrysolovay/Desktop/cdk-fargate-next-test/node_modules/next/dist/export/index.js:22:1166)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async build (/Users/harrysolovay/Desktop/cdk-fargate-next-test/node_modules/next/dist/build/index.js:36:218)`

**Comment by timneutkens** [comment]
Please add a complete reproduction.
